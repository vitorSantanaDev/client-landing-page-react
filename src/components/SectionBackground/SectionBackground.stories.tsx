import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionBackground from '.'

export default {
  title: 'SectionBackground',
  component: SectionBackground,
  args: {
    children: (
      <div>
        <h1>Section Background</h1>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Temporibus
          quae numquam illum delectus optio adipisci in repudiandae eligendi
          minus eum nihil aperiam saepe, porro aspernatur, dolore assumenda
          impedit. Quo, similique.
        </p>
      </div>
    )
  }
} as ComponentMeta<typeof SectionBackground>

export const LightOnly: ComponentStory<typeof SectionBackground> = (args) => (
  <SectionBackground {...args} />
)

export const DarkOnly: ComponentStory<typeof SectionBackground> = (args) => (
  <SectionBackground {...args} />
)

DarkOnly.args = {
  background: true
}
