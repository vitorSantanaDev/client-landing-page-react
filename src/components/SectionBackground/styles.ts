import styled, { css, DefaultTheme } from 'styled-components'

interface IWrapperProps {
  background?: boolean
}

const containerBackgroundActivate = (theme: DefaultTheme) => css`
  background-color: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`

export const Wrapper = styled.div<IWrapperProps>`
  ${({ theme, background }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    min-height: 100vh;

    display: flex;
    align-items: center;

    ${background && containerBackgroundActivate(theme)};
  `}
`
