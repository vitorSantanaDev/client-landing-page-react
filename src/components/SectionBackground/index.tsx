import React from 'react'

import SectionContainer from '../SectionContainer'
import * as S from './styles'

interface ISectionBackgroundProps {
  children: React.ReactNode
  background?: boolean
}

const SectionBackground: React.FC<ISectionBackgroundProps> = ({
  children,
  background
}) => (
  <S.Wrapper background={background}>
    <SectionContainer>{children}</SectionContainer>
  </S.Wrapper>
)

export default SectionBackground
