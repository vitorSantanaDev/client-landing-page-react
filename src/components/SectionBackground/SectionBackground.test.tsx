import React from 'react'
import { screen } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import SectionBackground from '.'

describe('<SectionBackground />', () => {
  it('should render with background dark', () => {
    const { container } = renderTheme(
      <SectionBackground background>
        <h1>Content</h1>
      </SectionBackground>
    )
    expect(screen.getByRole('heading')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render with background light', () => {
    const { container } = renderTheme(
      <SectionBackground>
        <h1>Content</h1>
      </SectionBackground>
    )
    expect(
      screen.getByRole('heading', { name: /content/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
