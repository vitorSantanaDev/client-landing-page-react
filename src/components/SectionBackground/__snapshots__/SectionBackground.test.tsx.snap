// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with background dark 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0A1128;
  color: #FFFFFF;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      Content
    </h1>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with background light 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      Content
    </h1>
  </div>
</div>
`;
